### ISSUE TITLE
There is no file of "monitor.html"

### ISSUE THREAD

**Issue by EddiePi** [intel-hadoop/HiBench#220]
I have run several HiBench data sets. In the directory `./report/<workload>/<language APIs>/`, there are only `bench.log` and `conf/` but without `monitor.html`. How can I get the file `monitor.html`?

**Comment by FedeSkia** [comment]
Hi, I have the same problem. Could someone help please? Thanks

**Comment by carsonwang** [comment]
It uses SSH to connect to other nodes and collect the data. It requires ability to SSH to other nodes without password. This usually works if HiBench is on the master node.

**Comment by EddiePi** [comment]
@carsonwang SSH access to other nodes without password is already configured. HiBench is on the master node of Yarn. I am also using the yarn-client mode on Spark. It still does not work.

**Comment by FedeSkia** [comment]
Hi @EddiePi. Did you solve? Thanks

**Comment by EddiePi** [comment]
@FedeSkia I did not solve it yet.

**Comment by FedeSkia** [comment]
@EddiePi can you find the "HiBench" application's in the history server? Because I cant. 
I'm reading: http://spark.apache.org/docs/1.3.0/monitoring.html and I think that the problem is due to HiBench cannot output the application event log... What do you think?
Thanks

**Comment by EddiePi** [comment]
@FedeSkia Yes! I have the same problem. I just want to review the job history but both approaches (history server and monitor.html) are not available.

**Comment by carsonwang** [comment]
Do you see the job history if you run Spark shell? 
HiBench doesn't use the spark-defaults.conf in spark/conf directory when submitting the application. So please set history server related configuration to HiBench's conf/99-user_defined_properties.conf.

There is a known issue for the HiBench monitoring. If there is a SSH Banner set, the monitoring scripts won't work. It uses python scripts to collect the system utilization data. What is your OS? I am wondering if this is also related to your OS environment.

**Comment by EddiePi** [comment]
Hi @FedeSkia I set history server configuration to 99-user_defined_properties.conf. But there is still no history log. Would you please paste your configuration file please.

Hi @carsonwang I can see the job history when I run Spark shell. SSH banner is not set. My OS is Ubuntu 15.10.

**Comment by FedeSkia** [comment]
Hi @EddiePi, 
Add this to the end of 99-user :
spark.eventLog.dir=
spark.eventLog.enabled=true
spark.yarn.historyServer.address=

**Comment by EddiePi** [comment]
@FedeSkia Thanks for your help. I did not add the eventLog configuration. It works now!

**Comment by FedeSkia** [comment]
@EddiePi @carsonwang Hi guys.
I m still stuck with the monitor.html file. Did you find a solution?
Thanks

**Comment by carsonwang** [comment]
@FedeSkia , you may have a try running bin/functions/monitor.py manually. You have to pass the required parameters as it shows the usage. Specify the parent pid and then kill the parent process. Please check if a log file is generated. Otherwise you may debug what's going wrong when executing it.
You can also use a 3rd party tool to monitor the system resource usage if this still doesn't work in your environment.

**Comment by vijaynirm** [comment]
Monitor.html not generating. In my case, it might be due to SSh banner set. Its enterprise application. any other way, whether we still get the monitor.Html, as i ma trying to find the thoughput of individual nodes across clusters.

**Comment by kishoregali** [comment]
Hi I am facing the same issue that monitor.html is not generated in my case and trying to run join benchmark individually .. but no monitor.html is created..

following info has been created
====================
 conf
rankings_uservisits_join.hive
bench.log


can somebody help me please

**Comment by vijaynirm** [comment]
It didnt work for me.....

**Comment by AdrienPIZZO** [comment]
Hi,

I'm stuck on the same problem, the workload run properly (i.e kmeans & wordcount),
all reports are generated except monitor.html,
logs in report aren't explicit for this issue

i would appreciate some help

**Comment by jairsonrodrigues** [comment]
I have spent a lot of time on this issue too.

**I solved it** with SSH passwordless method, as suggested above by @carsonwang on Jun 15, 2016. 

Notice that he said: 

> ... this USUALLY works if HiBench is on the master node.

Perhaps, in some scenarios, it is possible to face this problem in even though in master node. In this case, your instances do not allow the SSH passwordless between master and data nodes.

You must:

1) generate public and private keys of your master node
2) log out master node, and from another environment, you must remotely access the data node 1, 2, 3, etc.
3) in each of the data nodes, paste the content of your recently generated master node's public key into ~/.ssh/authorized_keys

That's all. After that, your master node can connect via SSH without password.

And Hibench will work.

Details and commands in [https://www.debian.org/devel/passwordlessssh](https://www.debian.org/devel/passwordlessssh)

**Comment by fleandr** [comment]
Another problem source: 
set up these parameters in hibench.conf
hibench.masters.hostnames
hibench.slaves.hostnames

You can set up $SLAVES variable as well

Problem is in these lines of monitor.py (514-516):
nodes = sorted(list(set(nodes)))
for node in nodes:
na.append(P(node, PROBE_INTERVAL))

If you do not have any nodes as slaves you will not get monitoring data and logs and html

check this by checking ps -afx | grep monitor.py whole running HiBench
It should have your hosts to be monitors as last arguments.
